### ISSUE TITLE
Legacy DRM API removal disables TTY usage

### ISSUE THREAD

**Issue by ag-eitilt** [mpv-player/mpv#12295]
### Important Information

Provide following Information:
- mpv 0.36.0
- Gentoo, kernel 6.1.41
- Official repository ebuild
- 0.36.0 -- specifically ba20f60addaa4ebdd9f6be25464c1b46a7f1f576
- None (tty)
- nouveau; can't find a version

### Reproduction steps

`mpv --vo=drm ...`

### Expected behavior

Technically, this is the expected behaviour according to the state of the code; however, it winds up breaking my use case.  The behaviour *I* would expect is for MPV to play the video successfully over the terminal.

### Actual behavior

MPV refuses to start as DRM Atomic instructions are not supported.

### Log file

[drm.txt](https://github.com/mpv-player/mpv/files/12470271/drm.txt)

### Discussion

I fully appreciate that the legacy API adds complexity, and that in general complexity is undesirable when it gets in the way of maintenance.  However, I make use of the bare TTYs very frequently in my daily work, and have really appreciated MPV being lightweight enough that I don't have to open a WM just to watch something quick.  I suppose the answer might just be for me to switch to `nvidia-drivers` (critically, nothing newer than 470.xx since I have an old card; how new is Atomic?) off of `nouveau`, but even if it's not a *very* common use case, I'm [not the only one](https://bbs.archlinux.org/viewtopic.php?id=288091) running into the problem.  How much maintenance burden do those 130 lines of code add?, because if it's indeed non-trivial then I accept dropping my niche use, but if it's just theoretical complexity then I'd feel a bit put out.

**Comment by CounterPillow** [comment]
What's stopping nouveau from supporting the atomic DRM API, aside from it being a borderline unmaintained driver?

**Comment by Dudemanguy** [comment]
The atomic API is like 10 years old or so. The code here was rewritten not too long after that commit, so it's not something that's just trivial.

**Comment by CounterPillow** [comment]
From `#dri-devel` on OFTC:
```
<CounterPillow> Does implementing the DRM atomic API require specific hardware support or is the lack of its implementation in the select few older drivers caused by a lack of developer time? mpv recently removed its support for the legacy DRM API, and now they have a nouveau user complaining.
<airlied> CounterPillow: lack of developer time, radeon and nouveau probably being the two main ones
<airlied> nouveau.atomic=1 might work around it, might also explode
<CounterPillow> Alright, thanks!
```

**tl;dr:** try `nouveau.atomic=1` in your kernel command line, and otherwise, start contributing to the kernel since you care about the hardware (or hire someone to work on this (or buy a new old GPU which is the cheapest option a used RX580 is like $60)).

**Comment by ag-eitilt** [comment]
> The atomic API is like 10 years old or so. The code here was rewritten not too long after that commit, so it's not something that's just trivial.

Fair enough, that's a good reason to not support it.

> From `#dri-devel` on OFTC:
> 
> ```
> <CounterPillow> Does implementing the DRM atomic API require specific hardware support or is the lack of its implementation in the select few older drivers caused by a lack of developer time? mpv recently removed its support for the legacy DRM API, and now they have a nouveau user complaining.
> <airlied> CounterPillow: lack of developer time, radeon and nouveau probably being the two main ones
> <airlied> nouveau.atomic=1 might work around it, might also explode
> <CounterPillow> Alright, thanks!
> ```
> 
> **tl;dr:** try `nouveau.atomic=1` in your kernel command line, and otherwise, start contributing to the kernel since you care about the hardware (or hire someone to work on this (or buy a new old GPU which is the cheapest option a used RX580 is like $60)).

Thanks, I'll take a look into that option!  (I really should upgrade quite a number of things on the computer, but in the mean time...)

**Comment by m040601** [comment]
> ... I make use of the bare TTYs very frequently in my daily work, 

Me too. I have been doing this for years. Had a nice setup with mpv, tmux and lotz of customizations for switching TTY's to watch video.

Until mpv stopped working on the console this year. I'm on Archlinux, have a 10 year old PC

```
                description: VGA compatible controller
                product: RS780 [Radeon HD 3200]
                vendor: Advanced Micro Devices, Inc. [AMD/ATI]
```

with the Xorg amd ati driver for older GPU's (not the Xorg amdgpu for the recent ones)

```
local/xf86-video-ati 1:22.0.0-1 (xorg-drivers)
    X.org ati video driver
```

All the "mesa's" and "vulkan's" and "hw acceleration" that a end user is supposed to install. No hacking, no messing with configs.

Everything worked fine until this year.

I still havent figured out exactly what happened.

Is my hardware "too old" ?

> ....and have really appreciated MPV being lightweight enough 
> that I don't have to open a WM just to watch something quick.

Yeap me too. We are in a minority nowadays I suppose.

**Comment by CounterPillow** [comment]
@m040601 
Your hardware isn't too old, it's just that nobody cares enough about the driver it uses to update it to the new atomic API that has been around for a decade. You can be the person that changes this, if you still care about the hardware.
